tools/Evm/T8n/JsonConverters/ReceiptJsonConverter.cs

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Int256;
+using Nethermind.Serialization.Json;
+
+namespace Evm.T8n.JsonConverters;
+
+// required to serialize in geth t8n format
+public class ReceiptJsonConverter : JsonConverter<TxReceipt>
```

Comment:
Can we achieve Geth format without converter? What about `ReceiptForRpc`?

Reply:
Can you explain in detail what are the differences? Can we just inherit from it and change something?

Reply:
```
type Receipt struct {
	// Consensus fields: These fields are defined by the Yellow Paper
	Type              uint8  `json:"type,omitempty"`
	PostState         []byte `json:"root"`
	Status            uint64 `json:"status"`
	CumulativeGasUsed uint64 `json:"cumulativeGasUsed" gencodec:"required"`
	Bloom             Bloom  `json:"logsBloom"         gencodec:"required"`
	Logs              []*Log `json:"logs"              gencodec:"required"`

	// Implementation fields: These fields are added by geth when processing a transaction.
	TxHash            common.Hash    `json:"transactionHash" gencodec:"required"`
	ContractAddress   common.Address `json:"contractAddress"`
	GasUsed           uint64         `json:"gasUsed" gencodec:"required"`
	EffectiveGasPrice *big.Int       `json:"effectiveGasPrice"` // required, but tag omitted for backwards compatibility
	BlobGasUsed       uint64         `json:"blobGasUsed,omitempty"`
	BlobGasPrice      *big.Int       `json:"blobGasPrice,omitempty"`

	// Inclusion information: These fields provide information about the inclusion of the
	// transaction corresponding to this receipt.
	BlockHash        common.Hash `json:"blockHash,omitempty"`
	BlockNumber      *big.Int    `json:"blockNumber,omitempty"`
	TransactionIndex uint        `json:"transactionIndex"`
}
```
here omitempty means to omit a field from being included in the JSON output if the field has an empty value

nethermind receipts:
```
[
  {
    "transactionHash": "0xa98a24882ea90916c6a86da650fbc6b14238e46f0af04a131ce92be897507476",
    "transactionIndex": "0x0",
    "blockNumber": "0x0",
    "cumulativeGasUsed": "0x84d0",
    "gasUsed": "0x84d0",
    "from": "0xd02d72e067e77158444ef2020ff2d325f929b363",
    "to": "0x1111111111111111111111111111111111111111",
    "contractAddress": null,
    "logs": [],
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "status": "0x0",
    "error": "BadInstruction",
    "type": "0x2"
  },
  {
    "transactionHash": "0x36bad80acce7040c45fd32764b5c2b2d2e6f778669fb41791f73f546d56e739a",
    "transactionIndex": "0x1",
    "blockNumber": "0x0",
    "cumulativeGasUsed": "0x109a0",
    "gasUsed": "0x84d0",
    "from": "0xd02d72e067e77158444ef2020ff2d325f929b363",
    "to": "0x1111111111111111111111111111111111111111",
    "contractAddress": null,
    "logs": [],
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "status": "0x0",
    "error": "BadInstruction",
    "type": "0x2"
  }
]
```
geth receipts:
```
[
  {
    "type": "0x2",
    "root": "0x",
    "status": "0x0",
    "cumulativeGasUsed": "0x84d0",
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "logs": null,
    "transactionHash": "0xa98a24882ea90916c6a86da650fbc6b14238e46f0af04a131ce92be897507476",
    "contractAddress": "0x0000000000000000000000000000000000000000",
    "gasUsed": "0x84d0",
    "effectiveGasPrice": null,
    "blockHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "transactionIndex": "0x0"
  },
  {
    "type": "0x2",
    "root": "0x",
    "status": "0x0",
    "cumulativeGasUsed": "0x109a0",
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "logs": null,
    "transactionHash": "0x36bad80acce7040c45fd32764b5c2b2d2e6f778669fb41791f73f546d56e739a",
    "contractAddress": "0x0000000000000000000000000000000000000000",
    "gasUsed": "0x84d0",
    "effectiveGasPrice": null,
    "blockHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "transactionIndex": "0x1"
  }
]
```

Reply:
Looks close enough to reuse it with some additional instrumentation? Maybe you can tackle it in the 2nd PR.